lib/lambda-worker/lambda-worker.ts

```diff
@@ -0,0 +1,199 @@
+
+    // Lambda settings
+    const memorySize =
+      props.lambdaProps.memorySize &&
+      props.lambdaProps.memorySize > MINIMUM_MEMORY_SIZE
+        ? props.lambdaProps.memorySize
+        : MINIMUM_MEMORY_SIZE;
+
+    const lambdaTimeout =
+      props.lambdaProps.timeout &&
+      props.lambdaProps.timeout.toSeconds() > MINIMUM_LAMBDA_TIMEOUT.toSeconds()
+        ? props.lambdaProps.timeout
+        : MINIMUM_LAMBDA_TIMEOUT;
+
+    // Queue settings
+    const maxReceiveCount = props.queueProps.maxReceiveCount
+      ? props.queueProps.maxReceiveCount
+      : DEFAULT_MAX_RECEIVE_COUNT;
+
+    const queueTimeout = cdk.Duration.seconds(
+      maxReceiveCount * lambdaTimeout.toSeconds()
+    );
+
+    const approximateAgeOfOldestMessageThreshold = props.queueProps
+      .approximateAgeOfOldestMessageThreshold
+      ? props.queueProps.approximateAgeOfOldestMessageThreshold
+      : DEFAULT_APPROX_AGE_OLDEST_MESSAGE_THRESHOLD;
+
+    const approximateNumberOfMessagesVisibleThreshold = props.queueProps
+      .approximateNumberOfMessagesVisibleThreshold
+      ? props.queueProps.approximateNumberOfMessagesVisibleThreshold
+      : DEFAULT_APPROX_NUM_MESSAGES_VISIBLE_THRESHOLD;
+
+    // Topic Subscription Settings
+    let subscriptionProps = {};
+    if (props.filterPolicy) {
+      subscriptionProps = { filterPolicy: props.filterPolicy };
+    }
+
+    // Create both the main queue and the dead letter queue
+    const lambdaDLQ = new sqs.Queue(this, `${props.name}-dlq`, {
+      queueName: `${props.name}-dlq`,
+      visibilityTimeout: queueTimeout,
+    });
+
+    const lambdaQueue = new sqs.Queue(this, `${props.name}-queue`, {
+      queueName: `${props.name}-queue`,
+      visibilityTimeout: queueTimeout,
+      deadLetterQueue: { queue: lambdaDLQ, maxReceiveCount: maxReceiveCount },
+    });
+
+    // If we have specified a topic, then subscribe
+    // the main queue to the topic.
+    if (props.topic) {
+      props.topic.addSubscription(
+        new subs.SqsSubscription(lambdaQueue, subscriptionProps)
+      );
+    }
+
+    // Create the lambda
+    const lambdaWorker = new lambdaNodeJs.NodejsFunction(
+      this,
+      `${props.name}`,
+      {
+        functionName: `${props.name}`,
+
+        // Pass through props from lambda props object
+        // Documented here https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-lambda-nodejs.NodejsFunctionProps.html
+        entry: props.lambdaProps.entry,
+        handler: props.lambdaProps.handler,
+        description: props.lambdaProps.description,
+        environment: props.lambdaProps.environment,
+        reservedConcurrentExecutions:
+          props.lambdaProps.reservedConcurrentExecutions,
+        retryAttempts: props.lambdaProps.retryAttempts,
+        role: props.lambdaProps.role,
+        securityGroup: props.lambdaProps.securityGroup,
+        timeout: lambdaTimeout,
+        vpc: props.lambdaProps.vpc,
+        vpcSubnets: props.lambdaProps.vpcSubnets,
```

Comment:
Could you achieve this with:

```js
{
    ...lambdaProps,
    functionName: `${props.name}`,
}
```

etc.

Reply:
I think you could achieve it with that, which would make sense if we did thrown an error for invalid properties. But if we do stick with the current method of setting them to minimum values then I think this approach will cause less confusion.

Reply:
Can we record the context for that decision here, @ShaunParsons ?